app/models/subscription.rb

```diff
@@ -238,8 +238,11 @@ def adjusted_boundaries(datetime, boundaries)
       to_datetime: dates_service.to_datetime,
       charges_from_datetime: dates_service.charges_from_datetime,
       charges_to_datetime: dates_service.charges_to_datetime,
+      fixed_charges_from_datetime: dates_service.fixed_charges_from_datetime,
+      fixed_charges_to_datetime: dates_service.fixed_charges_to_datetime,
       timestamp: datetime,
-      charges_duration: dates_service.charges_duration_in_days
+      charges_duration: dates_service.charges_duration_in_days,
+      fixed_charges_duration: dates_service.fixed_charges_duration_in_days
     )
 
     InvoiceSubscription.matching?(self, previous_period_boundaries) ? boundaries : previous_period_boundaries
```

Comment:
if we start populating this values, will it break? (comparing new invoice with stored in the db)

Reply:
cancel all the worries, it only applies if "fixed_charges_billed_in_monthly_split_intervals" - plan is yearly, fixed_charges are applied monthly 😌